projects/plugins/jetpack/extensions/store/membership-products/test/actions-test.js

```diff
@@ -0,0 +1,210 @@
+            const paymentPlanProductType = PRODUCT_TYPE_PAYMENT_PLAN;
+            const noticeMock = jest.spyOn( utils, 'onError' ).mockImplementation( () => {} );
+            const getMessageMock = jest.spyOn( message, 'getMessageByProductType' ).mockImplementation(() => {});
+
+            // When
+            await saveProduct( testCase.product, paymentPlanProductType, selectedProductIdCallback )(() => {}, () => {});
+
+            // Then
+            expect( noticeMock ).toBeCalled();
+            expect( getMessageMock ).toBeCalledWith( 'product requires a name', paymentPlanProductType );
+        });
+    });
+
+    test( 'having a price below the minimum price triggers an error notice.', async () => {
+        // Given
+        const ANY_MINIMUM_CURRENCY_AMOUNT = 123;
+        const productWithValueBelowMin = buildAnyValidProduct();
+        const paymentPlanProductType = PRODUCT_TYPE_PAYMENT_PLAN;
+        productWithValueBelowMin.price = ANY_MINIMUM_CURRENCY_AMOUNT - 1;
+        const noticeMock = jest.spyOn( utils, 'onError' ).mockImplementation( () => {} );
+        jest.spyOn( currencies, 'minimumTransactionAmountForCurrency' )
+            .mockImplementation( () => ANY_MINIMUM_CURRENCY_AMOUNT );
+
+        // When
+        await saveProduct( productWithValueBelowMin, paymentPlanProductType, () => {} )(() => {}, () => {});
+
+        // Then
+        expect( noticeMock ).toBeCalled();
+    } );
+
+    test( 'an invalid product price triggers an error notice.', async () => {
+        // Given
+        const productWithInvalidPrice = buildAnyValidProduct();
+        productWithInvalidPrice.price = 'anyInvalidPrice';
+        const paymentPlanProductType = PRODUCT_TYPE_PAYMENT_PLAN;
+        const noticeMock = jest.spyOn( utils, 'onError' ).mockImplementation( () => {} );
+        const getMessageMock = jest.spyOn( message, 'getMessageByProductType' ).mockImplementation(() => {});
+
+        // When
+        await saveProduct( productWithInvalidPrice, paymentPlanProductType, () => {} )(() => {}, () => {});
+
+        // Then
+        expect( noticeMock ).toBeCalled();
+        expect( getMessageMock ).toBeCalledWith( 'product requires a valid price', paymentPlanProductType );
+    });
+
+    /*test( 'Happy case displays a success notice.', async () => {
+        // Given
+        const anyValidProduct = buildAnyValidProduct();
+        const paymentPlanProductType = PRODUCT_TYPE_PAYMENT_PLAN;
+        const selectedProductCallback = jest.fn(() => {} );
+        const apiResponseProduct = {
+            id: 1,
+            title: 'anyTitle',
+            interval: 'anyInterval',
+            price: '12',
+            currency: 'anyCurrency',
+        }
+        const registryProductList = [ apiResponseProduct, apiResponseProduct ];
+        const registry = {
+            select: () => ( { getProducts: () => registryProductList } )
+        };
+        const dispatch = jest.fn( () => {} );
```

Comment:
Yes, having a high number of mocks and spies is a smell that the code could be refactored to more components. In this first part, I'm more focused on having test coverage rather than having testable code.